core/app/models/spree/order_taxation.rb

```diff
@@ -0,0 +1,83 @@
+module Spree
+  # Relatively simple class used to apply a {Spree::Tax::TaxedOrder} to a
+  # {Spree::Order}.
+  #
+  # This class will create or update adjustments on the taxed items and remove
+  # any now inapplicable tax adjustments from the order.
+  class OrderTaxation
+    # Create a new order taxation.
+    #
+    # @param [Spree::Order] order the order to apply taxes to
+    # @return [Spree::OrderTaxation] a {Spree::OrderTaxation} object
+    def initialize(order)
+      @order = order
+    end
+
+    # Apply taxes to the order.
+    #
+    # This method will create or update adjustments on all line items and
+    # shipments in the order to reflect the appropriate taxes passed in. It
+    # will also remove any now inapplicable tax adjustments.
+    #
+    # @param [Spree::Tax::TaxedOrder] taxes the taxes to apply to the order
+    # @return [void]
+    def apply(taxes)
+      @order.line_items.each do |item|
+        rates = taxes.line_item_taxes.select { |i| i.id == item.id }
+        update_adjustments(item, rates)
+      end
+
+      @order.shipments.each do |item|
+        rates = taxes.shipment_taxes.select { |i| i.id == item.id }
+        update_adjustments(item, rates)
+      end
+    end
+
+    private
+
+    # Walk through the taxes for an item and update adjustments for it. Once
+    # all of the taxes have been added as adjustments, remove any old tax
+    # adjustments that weren't touched.
+    #
+    # @private
+    # @param [#adjustments] item a {Spree::LineItem} or {Spree::Shipment}
+    # @param [Array<Spree::Tax::TaxedItem>] rates a list of calculated taxes for an item
+    # @return [void]
+    def update_adjustments(item, rates)
```

Comment:
Can we leave the rates out of the signature of this method, and rather attach them to the `TaxedItem` object? Because external services do not have actual rates.